ui/pages/confirmations/components/confirm/info/typed-sign/permit-simulation/permit-simulation.tsx

```diff
@@ -53,34 +54,37 @@ const PermitSimulation: React.FC<object> = () => {
       title={t('simulationDetailsTitle')}
       titleTooltip={t('simulationDetailsTitleTooltip')}
       description={t('permitSimulationDetailInfo')}
-      simulationHeading={t('spendingCap')}
       simulationElements={
-        Array.isArray(tokenDetails) ? (
-          <Box
-            display={Display.Flex}
-            flexDirection={FlexDirection.Column}
-            gap={2}
-          >
-            {tokenDetails.map(
-              (
-                { token, amount }: { token: string; amount: string },
-                i: number,
-              ) => (
-                <PermitSimulationValueDisplay
-                  key={`${token}-${i}`}
-                  primaryType={primaryType}
-                  tokenContract={token}
-                  value={amount}
-                />
-              ),
+        <ConfirmInfoRow label={t('spendingCap')}>
+          <Box style={{ marginLeft: 'auto', maxWidth: '100%' }}>
+            {Array.isArray(tokenDetails) ? (
+              <Box
+                display={Display.Flex}
+                flexDirection={FlexDirection.Column}
+                gap={2}
+              >
+                {tokenDetails.map(
+                  (
+                    { token, amount }: { token: string; amount: string },
+                    i: number,
+                  ) => (
+                    <PermitSimulationValueDisplay
```

Comment:
Minor, since there is lots of nesting here, is it worth separate `Token` and `SpendingCapRow` components in this file?